### ISSUE TITLE
Allow transactions to be filtered based on attachments

### ISSUE THREAD

**Issue by tactilis** [moneymanagerex/moneymanagerex#5697]
At present, if a transaction has file attachments added, then '[Att.]' is prepended to any user entered Notes to signify this fact.

An Attachment is a different attribute of a transaction to a Note and, IMO, the two should not be combined just for the sake of expediency.

One consequence of the present implementation is that it's not possible to find all the transactions with (or without) attachments by sorting the Notes column to group those having [Att] because, quite correctly, the Notes column sorts on the actual text the user has entered, not the presence of [Att.]

![image](https://user-images.githubusercontent.com/4060092/220475430-2e736151-9086-47ef-977d-a513205a593b.png)

I propose a dedicated `Attachments` column be added to MMEX. The data for each transaction is a count of the number of attachments (none = blank, or 1, 2, 3...).

This would allow transactions with attachments to easily be grouped together by sorting on the Attachments column.

Rejected idea: When initially discussed, I suggested the use of a paperclip symbol in the header of the Attachments column. On reflection, it would be better to use the word Attachments, which is consistent with the style of the other column headers. The user can resize the column down to `Att...` if they don't need to see the whole word.

The default position of the Attachments column should be immediately after Notes.

**Comment by vomikan** [comment]
In this case, it would be possible to add another menu item to display only transactions with an attachment.
![image](https://github.com/moneymanagerex/moneymanagerex/assets/6836805/4d2c294f-e9d8-4202-9ce1-6a31762e9669)

Or checkbox here
![image](https://github.com/moneymanagerex/moneymanagerex/assets/6836805/1c4067a1-24c2-449c-a282-4b2721ab5527)

**Comment by whalley** [comment]
Yes, I prefer adding a new row to the filter that allows transactions without or with attachements to be filtered.

Moved to v1.7.2 for development, as we are looking to get v1.7.1 our very soon and trying to limit to bug fixes.

**Comment by luisdev** [comment]
This still deserves consideration.
